File: pset1/mario2.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    for (int i = 1; i <= n; ++i)
    {
        // check the boundray
        for (int j = 1; j <= n + i + 2; ++j)
        {
            // check the two parts
            if ((j >= n + 1 - i && j >= 1 && j <= n) || (j >= n + 3 && j <= 2 * n + 2 && j <= i + n + 2))
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }
        printf("\n");
    }
}

File: pset6/mario2/mario2.py
from cs50 import get_int


def main():
    n = getProperInt()
    for i in range(1, n + 1):
        for j in range(1, n + i + 3):
            if (j >= n + 1 - i and j >= 1 and j <= n) or (j >= n + 3 and j <= 2 * n + 2 and j <= i + n + 2):
                print("#", end="")
            else:
                print(" ", end="")
        print()


def getProperInt():
    while True:
        n = get_int("Height: ")
        if n > 0 and n < 9:
            break
    return n


main()
